modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptRxjsClientCodegen.java

```diff
@@ -190,22 +201,79 @@ private void addOperationModelImportInfomation(Map<String, Object> operations) {
     }
 
     private void updateOperationParameterEnumInformation(Map<String, Object> operations) {
-        // This method will add extra infomation as to whether or not we have enums and
+        // This method will add extra information as to whether or not we have enums and
         // update their names with the operation.id prefixed.
         Map<String, Object> _operations = (Map<String, Object>) operations.get("operations");
-        List<CodegenOperation> operationList = (List<CodegenOperation>) _operations.get("operation");
-        boolean hasEnum = false;
-        for (CodegenOperation op : operationList) {
+        List<ExtendedCodegenOperation> operationList = (List<ExtendedCodegenOperation>) _operations.get("operation");
+        boolean hasEnums = false;
+        for (ExtendedCodegenOperation op : operationList) {
             for (CodegenParameter param : op.allParams) {
                 if (Boolean.TRUE.equals(param.isEnum)) {
-                    hasEnum = true;
+                    hasEnums = true;
                     param.datatypeWithEnum = param.datatypeWithEnum
                             .replace(param.enumName, op.operationIdCamelCase + param.enumName);
                 }
             }
         }
 
-        operations.put("hasEnums", hasEnum);
+        operations.put("hasEnums", hasEnums);
+    }
+
+    private void addConditionalImportInformation(Map<String, Object> operations) {
```

Comment:
could this be moved to the abstract typescript codegen?

Reply:
I suppose it could go there so other generators can use it as well. 
Should I move it there?

Reply:
yes, please do that

Reply:
This would require moving `ExtendedCodegenOperation` also into `AbstractTypeScriptClientCodegen` because of the newly added `hasHttpHeaders` field. I have the feeling that doing so would impact the other generators using the basic `CodegenOperation`?

Reply:
I can't really tell you the impact of that, but maybe we leave it as it is and consider it in the typescript refactoring.

Reply:
Ok, great. Then I leave it as it is.